packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -223,6 +238,37 @@ export class Outbox {
 		}
 	}
 
+	public submitBlobAttach(message: BatchMessage) {
+		this.maybeFlushPartialBatch();
+
+		if (
+			!this.blobAttachBatch.push(
+				message,
+				this.isContextReentrant(),
+				this.params.getCurrentSequenceNumbers().clientSequenceNumber,
+			)
+		) {
+			throw new GenericError("BatchTooLarge", /* error */ undefined, {
+				opSize: message.contents?.length ?? 0,
+				batchSize: this.blobAttachBatch.contentSizeInBytes,
+				count: this.blobAttachBatch.length,
+				limit: this.blobAttachBatch.options.hardLimit,
+			});
+		}
```

Comment:
I'm curious if extracting this in a separate (private) method and reusing it on the other batch managers would add a bit less overhead to the bundle size. The code below is only for 'attach' messages so it doesn't pass the 'Rule of three' 😄